tuf/api/metadata.py

```diff
@@ -1301,3 +1301,30 @@ def to_dict(self) -> Dict[str, Any]:
     def update(self, fileinfo: TargetFile) -> None:
         """Assigns passed target file info to meta dict."""
         self.targets[fileinfo.path] = fileinfo
+
+    def add_key(self, role: str, key: Key) -> None:
+        """Adds new signing key for delegated role 'role'.
+
+        Raises:
+            KeyError: If there is no role 'role' in the Delegations object.
+        """
+        if self.delegations is None:
+            return
```

Comment:
We should'nt  silently ignore when the operation failed. I think an exception should be raised here.